### ISSUE TITLE
Incorrect lifetime for AAD in open_in_place

### ISSUE THREAD

**Issue by bdonlan** [briansmith/ring#767]
Currently, the `open_in_place` API takes AAD with the same lifetime as the plaintext/ciphertext buffer:

```
pub fn open_in_place<'a>(
    key: &OpeningKey, 
    nonce: Nonce, 
    aad: Aad<'a>, 
    in_prefix_len: usize, 
    ciphertext_and_tag_modified_in_place: &'a mut [u8]
) -> Result<&'a mut [u8], Unspecified>
```

There is no reason the AAD has to live as long as the returned plaintext reference, and so this signature should be changed to allow the AAD to have a shorter lifetime than the in/out buffer.

Note that this is a regression from 0.13.x, which used an implicit lifetime for the AAD buffer (as `&[u8]`): https://docs.rs/ring/0.13.5/ring/aead/fn.open_in_place.html

**Comment by briansmith** [comment]
Thanks for reporting this! Fixed in  0e80eeb2aa05615b3bab00af32c387c53115e6ac.

**Comment by briansmith** [comment]
Released in 0.14.3.
